backend/app/api/ws/device_namespace.py

```
@@ -1248,26 +1249,36 @@ async def _broadcast_device_upgrade_status(
+            target_user_ids = {user_id}
 
-            await sio.emit(
-                "device:upgrade_status",
-                event_data,
-                room=f"user:{user_id}",
-                namespace="/chat",
-            )
+            with _db_session() as db:
+                admin_user_ids = (
+                    db.query(User.id)
+                    .filter(User.role == "admin", User.is_active == True)
```

Ruff (0.15.7)

[error] Avoid equality comparisons to `True`; use `User.is_active:` for truth checks

Replace with `User.is_active`

(E712)